api/notifications/notifications.go

```diff
@@ -0,0 +1,202 @@
+package notifications
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/gorilla/websocket"
+
+	"github.com/Peripli/service-manager/pkg/query"
+	"github.com/Peripli/service-manager/pkg/types"
+	"github.com/Peripli/service-manager/pkg/util"
+	"github.com/Peripli/service-manager/storage"
+
+	"github.com/Peripli/service-manager/pkg/log"
+
+	"github.com/Peripli/service-manager/pkg/web"
+	"github.com/Peripli/service-manager/pkg/ws"
+)
+
+const (
+	lastKnownRevisionHeader     = "last_known_revision"
+	lastKnownRevisionQueryParam = "last_known_revision"
+)
+
+func (c *Controller) handleWS(req *web.Request) (*web.Response, error) {
+	user, _ := web.UserFromContext(req.Context())
+	notificationQueue, lastKnownRevision, err := c.notificator.RegisterConsumer(user)
+	if err != nil {
+		return nil, fmt.Errorf("could not register notification consumer: %v", err)
+	}
+
+	revisionKnownToProxy := 0
+	revisionKnownToProxyStr := req.URL.Query().Get(lastKnownRevisionQueryParam)
+	if revisionKnownToProxyStr != "" {
+		revisionKnownToProxy, err = strconv.Atoi(revisionKnownToProxyStr)
+		if err != nil {
+			c.unregisterConsumer(notificationQueue)
+
+			log.C(req.Context()).Errorf("could not convert string to number: %v", err)
+			return nil, &util.HTTPError{
+				StatusCode:  http.StatusBadRequest,
+				Description: fmt.Sprintf("invalid %s query parameter", lastKnownRevisionQueryParam),
+				ErrorType:   "BadRequest",
+			}
+		}
+	}
+
+	if revisionKnownToProxy > 0 {
+		resp, err := c.isRevisionFound(req.Context(), revisionKnownToProxyStr)
```

Comment:
Notification cleaner might be executed between this call and the list operation.
Maybe get all notifications between `revisionKnownToProxy` and `lastKnownRevision` inclusive. Then sort them by revision (this will be workaround until there is storage orderby functionality) and check if the first notification has revision `revisionKnownToProxy`. If not - 410. It is also a good idea to check for revisionKnownToProxy == 0. In that case return directly 410.